meson.build

```diff
@@ -0,0 +1,32 @@
+project('icu', 'c', 'cpp',
+  version: '67.1',
+  meson_version: '>=0.55.99',
+  default_options: 'cpp_std=c++17')
+
+#add_project_arguments('-DU_USING_ICU_NAMESPACE=1', language : 'cpp')
+#add_project_arguments('-DU_HAVE_STD_STRING=1', language : 'cpp')
+
+if host_machine.system() == 'windows'
+   add_project_arguments('-DU_PLATFORM_USES_ONLY_WIN32_API', language : 'cpp')
+   #add_project_arguments('-DU_PLATFORM_HAS_WIN32_API', language : 'cpp')
+   add_project_arguments('-DWIN32', '-DWIN64', '-D_MBCS', '/utf-8', language : 'cpp')
+   add_project_arguments('-DU_PLATFORM_USES_ONLY_WIN32_API', language : 'c')
+   #add_project_arguments('-DU_PLATFORM_HAS_WIN32_API', language : 'c')
+   add_project_arguments('-DWIN32', '-DWIN64', '-D_MBCS', '/utf-8', language : 'c')
+endif
```

Comment:
Could these be passed in once with `language: ['c', 'cpp']`?

What do the commented out defines do?

Reply:
I took those from the VS project shipped by upstream (which did not actually compile, but I digress). Then I kept removing unnecessary ones until it worked. Commented out ones are deleted now.